### DISCUSSION TITLE
Add wait (5 secs) after each page load / redirect

### DISCUSSION THREAD

**Discussion by joseleperez** [cypress-io/cypress#26042] [Questions and Help]
Hi! 

After loading each page within my tests, I need to set a .wait(5secs) before any other action is taken. 

I know how to do it with the first-page visit (by overloading the .visit command), but that doesn't apply to page loads after clicking a link or normal navigation automatically triggered.

I tried putting it on the window:load event listener, but adding a cy.wait, there triggers a promise inside the promise problem. 

The use case that I have hundreds of tests already implemented on our site, but now we have some async JS that might modify the site, and the shortest path for not modifying all tests is just waiting for that JS to be triggered.

Thanks!

**Comment by portyu9** [comment]
Hi @joseleperez 

I would not add a global 5-second delay after every page load/redirect.

`window:load` is also not a good place to enqueue a `cy.wait()` globally. Cypress commands are queued commands, and inserting them from arbitrary event/promise callbacks can produce exactly the command/promise error you encountered.

Instead, wait for the thing that tells you the new page is actually ready.

For example, if navigation loads data:

```js
cy.intercept('GET', '/api/page-data*').as('pageData')

cy.contains('a', 'Next').click()

cy.location('pathname').should('eq', '/next')
cy.wait('@pageData')
cy.get('[data-testid="page-ready"]').should('be.visible')
```

If there is no network request to synchronize with, assert on the URL or an element that appears when the destination page is usable:

```js
cy.get('a').contains('Next').click()

cy.location('pathname').should('eq', '/next')
cy.get('[data-testid="destination-content"]').should('be.visible')
```

Cypress will retry those assertions until they pass or the configured timeout expires, so this handles fast and slow page loads without making every test wait an unconditional five seconds.

If the application genuinely contains a five-second timer that is part of the behavior being tested, use `cy.wait(5000)` only around that specific transition rather than after every navigation.

Please mark this answer as accepted if it helped, thank you.
